### PR TITLE
SHERMAN and FAHMY Adagrams 

### PR DESCRIPTION
**Pull request by Peacegypsy**
# Adagrams
Congratulations! You're submitting your assignment.

## Comprehension Questions

Feature | Feedback
--- | ---
What are the components that make up a method? | definition, argument parameters, code block, return statement, end.
What are the advantages of using git when collaboratively working on one code base? |both parties are able to edit and see the others changes.
What kind of relationship did you and your pair have with the unit tests? | we ran them frequently adn used them to clarify the requirements and debug.
Does your code use any methods from the `Enumerable` mixin? If so, where and why was it helpful? | we used .map to create a new array which helped us fix a pointer error.
What was one method you and your pair used to debug code? |reading through the rake tests and stack traces.
What are two discussion points that you and your pair discussed when giving/receiving feedback from each other that you would be willing to share? | patience was good, we worked at a consistent pace and switched roles regularly. Explanations were clear and unbiased.

### CODE REVIEW

**Comment by dHelmgren** [comment]
# Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | Answers are kind of terse, they'd benefit
Both teammates contributed to the codebase | Yep yep!
Small commits with meaningful commit messages | Clear and small commits! nice work!
**Code Requirements** |
`draw_letters` method | The drawing works, easy to read.
Uses appropriate data structure to store the letter distribution | A hash doesn't do what we want in this scenario, see my comments.
All tests for `draw_letters` pass | Pass!
`uses_available_letters?` method | Yeah! Good work!
All tests for `uses_available_letters?` pass | Pass!
`score_word` method | Fancy! Yours is the first to_sym solution I've seen.
Uses appropriate data structure to store the letter scores | I love a good case statement!
All tests for `score_word` pass | Pass!
`highest_score_from` method | Most concise answer I've seen! Great job!
Appropriately handles edge cases for tie-breaking logic | It sure does!
All tests for `highest_score_from` pass | Pass! 
**Overall** | Aside from the snafu with the hash at the beginning, this is a really solid demonstration of your skills!

File: lib/adagrams.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    end
+
+    word_score += 8 if (7..10).include?(word.length)
+
+    return word_score
+  end
+
+  def highest_score_from(words)
+    best_word = {word: nil, score: 0}
+    words.each do |word|
```
Comment (dHelmgren) [inline lib/adagrams.rb:90]: ONE LOOP I LOVE IT

File: lib/adagrams.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    end
+    hand
+
+  end
+
+  def uses_available_letters?(input, letter_in_hand)
+    new_letters = letter_in_hand.map { |i| i }
+    input.split("").each do |letter|
+      if !new_letters.include? letter
+        return false
```
Comment (dHelmgren) [inline lib/adagrams.rb:51]: Good job returning when you know the outcome!

File: lib/adagrams.rb [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      X: 1,
+      Y: 2,
+      Z: 1
+    }
+
+    hand = []
+    tiles_drawn = 0
+    until tiles_drawn == 10
+      letter = alphabet.keys.sample
+      if alphabet[letter] > 0
```
Comment (dHelmgren) [inline lib/adagrams.rb:37]: Using a hash with sample gives an equal 1/26 chance to each letter, instead we need the chances to be weighted.
